client-app/shared/common/composables/useSlugInfo.ts

```diff
@@ -1,17 +1,62 @@
 import { computed, toValue } from "vue";
-import { useFetch } from "@/core/api/common";
-import { useLanguages } from "@/core/composables";
-import type { SlugInfoResultType } from "@/shared/common/types";
+import { getPage, getSlugInfo } from "@/core/api/graphql";
+import { globals } from "@/core/globals";
+import type { PageTemplate } from "@/shared/static-content";
 import type { MaybeRefOrGetter } from "vue";
 
-export function useSlugInfo(seoUrl: MaybeRefOrGetter<string>) {
-  const { currentLanguage } = useLanguages();
+const RESERVED_URLS = ["__index__home__page__"];
 
-  const url = computed(() => `/storefrontapi/slug/${toValue(seoUrl)}?culture=${currentLanguage.value.cultureName}`);
-  const { data: slugInfo, isFetching: loading } = useFetch(url, { refetch: true }).get().json<SlugInfoResultType>();
+export function useSlugInfo(seoUrl: MaybeRefOrGetter<string>, isReserved?: boolean) {
+  const { storeId, userId, cultureName } = globals;
+  const variables = computed(() => {
+    return {
+      storeId,
+      userId,
+      cultureName,
+      slug: toValue(seoUrl),
+    };
+  });
+
+  const { result, loading: slugLoading } = getSlugInfo(variables);
+
+  const slugInfo = computed(() => {
+    if (!isReserved && RESERVED_URLS.includes(toValue(seoUrl))) {
+      return null;
+    }
+    return result.value?.slugInfo;
+  });
+
+  const hasContent = computed(() => {
+    return slugInfo.value?.entityInfo?.objectType === "ContentFile";
+  });
+
+  const getPageParams = computed(() => {
+    return { id: slugInfo?.value?.entityInfo?.objectId || "", cultureName, storeId };
+  });
+
+  const { load: fetchContent, result: contentResult, loading: contentLoading } = getPage(getPageParams);
```

Comment:
1. Move global variables to `useGetPage`
2. Use `fetchContent` as `fetchContent({ id: ... })

Reply:
?